### PR TITLE
Adding Advance Options to Openstack Bakery stage setup

### PR DESCRIPTION
**Pull request by edwinavalos**
Lets you implement extended attributes for packer

### CODE REVIEW

File: app/scripts/modules/openstack/pipeline/stages/bake/openstackBakeStage.js [hunk 1]
```diff
@@ -85,6 +98,35 @@ module.exports = angular.module('spinnaker.openstack.pipeline.stage.bakeStage',
...
+              value: '',
+            };
+          }
+        }
+      }).result.then(function(extendedAttribute) {
+          $scope.stage.extendedAttributes[extendedAttribute.key] = extendedAttribute.value;
+      });
+    };
+
+    this.removeExtendedAttribute = function (key) {
```
Comment (danielpeach) [inline app/scripts/modules/openstack/pipeline/stages/bake/openstackBakeStage.js]: Extra space here (should be `function(key) { ... }`).

File: app/scripts/modules/openstack/pipeline/stages/bake/openstackBakeStage.js [hunk 2]
```diff
@@ -73,10 +79,17 @@ module.exports = angular.module('spinnaker.openstack.pipeline.stage.bakeStage',
...
         }
         $scope.viewState.roscoMode = SETTINGS.feature.roscoMode;
+        $scope.showAdvancedOptions = showAdvanced();
         $scope.viewState.loading = false;
       });
     }
 
+    function showAdvanced() {
+      let stage = $scope.stage;
+      return !!(stage.templateFileName || (stage.extendedAttributes && _.size(stage.extendedAttributes) > 0) ||
```
Comment (danielpeach) [inline app/scripts/modules/openstack/pipeline/stages/bake/openstackBakeStage.js]: Are the values other than `extendedAttributes` and `templateFileName` relevant to an Openstack bake stage?

File: app/scripts/modules/openstack/pipeline/stages/bake/openstackBakeStage.js [hunk 3]
```diff
@@ -85,6 +98,35 @@ module.exports = angular.module('spinnaker.openstack.pipeline.stage.bakeStage',
       });
     }
 
+    this.addExtendedAttribute = function() {
+      if (!$scope.stage.extendedAttributes) {
+           $scope.stage.extendedAttributes = {};
```
Comment (danielpeach) [inline app/scripts/modules/openstack/pipeline/stages/bake/openstackBakeStage.js]: Indentation is off here.

File: app/scripts/modules/openstack/pipeline/stages/bake/bakeStage.html [hunk 4]
```diff
@@ -35,13 +35,57 @@ <h4 ng-if="viewState.loading" class="text-center">
...
+        <table class="table table-condensed packed">
+          <thead>
+          <tr>
+            <th style="width:40%">Key</th>
+            <th style="width:60%">Value</th>
+            <th class="text-right">Actions</th>
+          </tr>
+          </thead>
+          <tbody>
+          <tr ng-repeat="(key,value) in stage.extendedAttributes">
```
Comment (danielpeach) [inline app/scripts/modules/openstack/pipeline/stages/bake/bakeStage.html]: Please add a space between `key,` and `value`.

File: app/scripts/modules/openstack/pipeline/stages/bake/bakeStage.html [hunk 5]
```diff
@@ -35,13 +35,57 @@ <h4 ng-if="viewState.loading" class="text-center">
...
+            <td>
+              <strong class="small">{{key}}</strong>
+            </td>
+            <td>
+              <input type="text" required ng-model="stage.extendedAttributes[key]" value="{{value}}"
```
Comment (danielpeach) [inline app/scripts/modules/openstack/pipeline/stages/bake/bakeStage.html]: I know this is taken from other `bakeStage` templates, but I don't think you need `value="{{value}}"`.

**Comment by danielpeach** [comment]
Could you please amend your commit message to match the conventions (https://www.spinnaker.io/community/contributing/submitting/#type)? E.g., `feat(openstack): add advanced options to bake stage config`.

**Comment by edwinavalos** [comment]
@danielpeach I did the rebase horribly wrong because I tried to use the built in update branch button and everything is FUBAR, so I'll try opening this up in a new branch....

**Pull request closed without merging**